workflows/models.py

```diff
@@ -671,33 +660,36 @@ def create_output_datasets(self, tag_name=None, update=False):
                 read_type=lane["read_type"],
             ))
 
+        bam_cell_ids = metadata_yaml["meta"]["cell_ids"]
         output_file_info = []
-        for cell_id, metadata in cell_metadata.items():
-            log.info('getting bam metadata for cell {}'.format(cell_id))
-
-            bam_filepath = metadata['bam']
-
-            for filepath in (bam_filepath, bam_filepath + '.bai'):
+        for cell_id in bam_cell_ids:
```

Comment:
This should use the following section in the input yaml:
```  bams:
    template: {cell_id}.bam
    instances:
      - cell_id: CELL_ID_1
      - cell_id: CELL_ID_2```
as per: https://github.com/shahcompbio/single_cell_pipeline#bams